mxnet-bot/LabelBotFullFunctionality/test_labelbot.py

```diff
@@ -52,6 +51,68 @@ def test_update_labels(self):
             self.lb.all_labels = ['sample_label', 'another_label', 'all_labels']
             self.assertTrue(self.lb.update_labels(issue_num=0, labels=['sample_label']))
 
+    # Tests for different kinds of user input
+    # Tests for spaces
+    def test_tokenize_frontSpace(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[   Sample Label]")
+        self.assertEqual(user_label, ['sample label'])
+
+    def test_tokenize_endSpace(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[ Sample Label      ]")
+        self.assertEqual(user_label, ['sample label'])
+
+    def test_tokenize_midSpace(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[Sample        Label]")
+        self.assertEqual(user_label, ['sample label'])
+
+    def test_tokenize_manyWordsSpace(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[This    is    a     sample    label]")
+        self.assertEqual(user_label, ['this is a sample label'])
+
+    # Tests for case-insensitive
+    def test_tokenize_upperCase(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[SAMPLE LABEL, ANOTHER LABEL, FINAL]")
+        self.assertEqual(user_label, ['sample label', 'another label', 'final'])
+
+    def test_tokenize_mixCase(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[sAmPlE LaBeL, AnOtHeR lAbEl, fInAl]")
+        self.assertEqual(user_label, ['sample label', 'another label', 'final'])
+
+    def test_tokenize_lowerCase(self):
+        user_label = LabelBot._tokenize(LabelBot.__class__, "[sample label, another label, final]")
+        self.assertEqual(user_label, ['sample label', 'another label', 'final'])
+
+    # Tests for parsing data from github comments
+    # Referencing @mxnet-label-bot from different places in the comment body
+    def test_parse_webhook_data_referencedAtEnd(self):
+        with open("testInputFiles/testAtEnd.json", "r") as fh:
+            token = ast.literal_eval(fh.read())
+            with patch.object(LabelBot, '_secure_webhook', return_value=True):
+                with patch.object(LabelBot, 'add_labels', return_value=True):
+                    self.lb.parse_webhook_data(token)
+
+    def test_parse_webhook_data_referencedAtStart(self):
+        with open("testInputFiles/testAtStart.json", "r") as fh:
+            token = ast.literal_eval(fh.read())
```

Comment:
Please don't use literal_eval to parse json. Use normal json parsing. See json module.

Reply:
There is JSON embedded within a dictionary in the file. The file isn't acutally valid json. This is meant to be to emulate the behavior of the data that gets fed into parse_webhook after receiving the payload from SQS. Ast literal is used to keep the type as a dictionary after reading from the file (which normally gives a stringified version of a dictionary). As well single quotes have to be converted to double quotes. These steps can be done manually but I thought it would be best practice to use a built-in supported module like ast.literal_eval. 
https://stackoverflow.com/questions/14347047/single-versus-double-quotes-in-json-loads-in-python
https://stackoverflow.com/questions/988228/convert-a-string-representation-of-a-dictionary-to-a-dictionary